decode.go

```diff
@@ -170,6 +171,15 @@ func (d *decoder) eval(target reflect.Value) {
 					target.SetBytes(data)
 				}
 			case vm.String:
+				if target.Type() == uuidType {
+					val, err := gouuid.FromString(frame.String)
+					if err != nil {
+						d.error(fmt.Errorf("runtime error: invalid input UUID: %w", err))
```

Comment:
I'm not sure that this counts as a runtime error - runtime errors here are internal errors, but this happens if the input is malformed.
```suggestion
						d.error(fmt.Errorf("invalid UUID in Avro encoding: %w", err))
```